src/components/ContactsList/index.js

```diff
@@ -33,31 +56,53 @@ export const withContacts = BaseComponent =>
     };
 
     async componentDidMount() {
-      const contactsRaw = await cozy.client
-        .fetchJSON("GET", "/data/io.cozy.contacts/_all_docs?include_docs=true")
-        .then(data => data.rows)
-        .catch(error => {
-          this.setState(state => ({ ...state, contacts: [], error }));
-        });
-      const contacts = (contactsRaw || [])
-        .map(row => ({
-          ...row.doc
-        }))
-        .filter(contact => contact.email);
+      const contacts = await cozyHTTP.getContacts();
```

Comment:
Now we use the API instead of implementing the `getContacts` needs in the component that is not responsible for.